microscopium/_util.py

```diff
@@ -1,6 +1,82 @@
 import numpy as np
 import numbers
 
+
+def generate_spiral(dim, direction, clockwise=False):
+    """Generate an dim * dim of sequental integers in a spiral pattern.
+
+    Many HCS systems use a spiral pattern to capture multiple fields in
+    for each well. This function generates a spiral pattern of integers
+    to be used by the montaging functions.
+
+    Parameters
+    ----------
+    dim : int
+        The dimension of the spiral. The function will return a dim * dim
+        array.
```

Comment:
Yeah, as long as spirals have the shape ``(dim, dim - 1)`` or ``(dim - 1, dim)`` and a sensible choice is chosen for the starting co-ordinate this algorithm will work fine (this should always be ``[floor(nrow / 2), floor(ncol / 2)]``. It gets a bit more complicated if you start looking at arrays of any shape and any starting co-ordinate because then you need to thinking about what happens when the "cursor" leaves the array bounds.